benchmark/Rust/Savina/src/concurrency/CigaretteSmoker.lf

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2020 TU Dresden
+ *
+ * See C++ benchmark for documentation.
+ *
+ * @author Christian Menard 
+ * @author Hannes Klein
+ * @author Clément Fournier
+ */
+
+target Rust {
+    build-type : RelWithDebInfo,
+    cargo-features: [ "cli" ],
+    cargo-dependencies: {
+        rand: {
+            version: "0.8",
+            features: ["small_rng"],
+        },
+        reactor_rt: {
+            features: ["parallel-runtime"]
+        },
+    },
+};
+
+import BenchmarkRunner from "../BenchmarkRunner.lf";
+
+reactor ArbiterReactor(num_rounds: usize(1000), num_smokers: usize(200)) {
+    
+    preamble {=
+        use rand::rngs::SmallRng;
+        use rand::Rng;
+        use rand::SeedableRng;
+    =}
+
+    state num_smokers(num_smokers);
+    state num_rounds(num_rounds);
+    state roundsSoFar: usize(0);
+    state random: rand::rngs::SmallRng({= {
+        let seed = num_rounds * num_smokers;
+        rand::rngs::SmallRng::seed_from_u64(seed as u64)
+    }=});
+    
+    output[num_smokers] startSmoking: usize;
+    
+    logical action chooseSmoker;
+    
+    reaction(startup) -> chooseSmoker {=
+        ctx.schedule(chooseSmoker, Asap);
+    =}
+    
+    reaction(chooseSmoker) -> startSmoking, chooseSmoker {=
+        // assume resources grabbed instantaneously and choose a random smoker to start smoking
+        let new_smoker_index = self.random.gen_range(0..self.num_smokers);
+        let busy_wait_period = self.random.gen_range(10usize..1010);
```

Comment:
This does not use the pseudo random number generator as provided in Savina (and implemented in C++), does it? I think using the same generator is really important for comparability of the results.

Reply:
This is pretty much the standard random library for Rust. When I ported benchmarks I implemented the PRNG from C++, which can be found [here](https://github.com/lf-lang/lingua-franca/blob/rust-more-benchmarks/benchmark/Rust/Savina/src/pseudo_random.rs).
When porting existing benchmarks to the new benchmark runner I will switch the RNGs out accordingly. For example, [here](https://github.com/lf-lang/lingua-franca/blob/rust-more-benchmarks/benchmark/Rust/Savina/src/parallelism/RadixSort.lf) is RadixSort, which I already ported. Performance almost doubled after switching.